### PR TITLE
FW LAUNCH: Apply surface deflection proportional to THR

### PR DESCRIPTION
**Pull request by fiam**
To avoid stalling the plane, start with a pitch angle of zero until
the launch is detected. Then, use a climb angle proportional to
the applied THR between the idle launch THR and the full launch THR.

References #2885

### CODE REVIEW

File: src/main/navigation/navigation.c [hunk 1]
```diff
@@ -132,18 +132,19 @@ PG_RESET_TEMPLATE(navConfig_t, navConfig,
...
-        .launch_max_angle = 45                 // 45 deg
+        .launch_idle_throttle = 1000,           // Motor idle or MOTOR_STOP
+        .launch_motor_timer = 500,              // ms
+        .launch_motor_spinup_time = 100,        // ms, time to gredually increase throttle from idle to launch
+        .launch_min_time = 0,                   // ms, min time in launch mode
+        .launch_timeout = 5000,                 // ms, timeout for launch procedure
+        .launch_max_altitude = 0,               // cm, altitude where to consider launch ended
+        .launch_climb_angle = 18,               // 18 degrees
+        .launch_max_angle = 45,                 // 45 deg
+        .launch_climb_angle_awaits_motor = true,
```
Comment (digitalentity) [inline src/main/navigation/navigation.c]: Please keep default to false for backward compatibility until the feature is better tested.

Comment (fiam) [reply src/main/navigation/navigation.c]: My intention is to eventually make this work better in all cases, so we can turn it on for everyone. However, we can ship it in a release sooner by turning it off by default, so I'll change it in the meantime.

Comment (digitalentity) [reply src/main/navigation/navigation.c]: Yes, I agree, we eventually should enable this. But I want more testing.

**Comment by fiam** [comment]
This was flight tested yesterday with bungee launch by @shellixyz and by hand throw by me.

- Bungee launch worked properly
- Hand launch sent the wing to the ground, broke the motor mount and I couldn't test anymore. I suspect that the problem is that I threw it at waist level and parallel to the ground (as I always do with immediate servo deflection), but with delayed servo deflection you need to launch it with some angle and higher (e.g. at head level).

**Review by digitalentity — APPROVED** [review]

**Comment by digitalentity** [comment]
@fiam code looks ok to me, however delayed servo deflection may be less safe after all.

**Comment by fiam** [comment]
Yep, looks like with hand launch  we need to pull up ASAP. Otherwise the vertical speed is too much by the time the motor starts and it can’t start climbing on time.

**Comment by Dronek** [comment]
I see some misconcenption here. Some pilots intentionally toss their aircraft at a steep angle to give it some altitude in order to account for any drop before the plane starts picking up speed. Restricting the pitch angle to zero in the beginning will counteract this intention and level the aircraft too early. If you need to launch your plane at a pitch angle of zero then just set launch climb angle to zero and pull up manually after the first few meters.

It might be more useful to disable any leveling at the beginning of the launch phase and start leveling (with applied climb angle) as soon as the motor starts. This way anyone can choose their own tossing angle and choose when the plane starts a controlled climb by adjusting the launch delay.

For example: 
- until plane is tossed: acro mode
- after plane is tossed and motor still idle: acro mode
- as soon as throttle starts increasing: angle mode with pitch up according to climb angle setting

**Pull request closed without merging**